src/prefect/core/flow.py

```diff
@@ -1133,6 +1133,11 @@ def serialize(self, build: bool = False) -> dict:
                 raise ValueError("This flow has no storage to build")
             if self.name not in self.storage:
                 self.storage.add_flow(self)
+            else:
+                warnings.warn(
+                    "This flow is already contained in storage; if you changed your Flow since"
```

Comment:
Small suggestion: The *strictly true* statement here is that "a flow with this name is already contained in storage". Since there might be times when one does, in fact, want to overwrite the flow by that name, perhaps this method should have a `force=False` or `always_build=False` kwarg that would allow calling `add_flow` even when the names match. Otherwise the warning is helpful but the user has to totally reinstantiate their Storage object to get around it. 

(not totally sure about what to call that kwarg, these are just suggestions)

Thoughts?

Reply:
Yea, I'm definitely open to it, the only "catch" is that I'll have to implement a `remove_flow` method on all storage objects to call here, so it feels like a heavy change (but that doesn't mean we shouldn't do it).

Reply:
I see, that makes sense and does feel superfluous. Perhaps the warning can just indicate that the user should reinstantiate the Storage object

Reply:
Updated